ethereum/contract/BREthereumToken.c

```diff
@@ -117,6 +117,59 @@ tokenGetContract(BREthereumToken token) {
     return contractERC20;
 }
 
+extern BREthereumHash
+tokenGetHash (BREthereumToken token) {
+    return addressGetHash(token->raw);
+}
+
+extern BREthereumToken
+tokenCreate (const char *address,
+             const char *symbol,
+             const char *name,
+             const char *description,
+             int decimals,
+             BREthereumGas defaultGasLimit,
+             BREthereumGasPrice defaultGasPrice) {
+    BREthereumToken token = malloc (sizeof(struct BREthereumTokenRecord));
+
+    token->address     = strdup (address);
+    token->symbol      = strdup (symbol);
+    token->name        = strdup (name);
+    token->description = strdup (description);
+    token->decimals    = decimals;
+    token->gasLimit    = defaultGasLimit;
+    token->gasPrice    = defaultGasPrice;
+    token->raw         = addressCreate (address);
+
+    return token;
+}
+
+extern void
+tokenRelease (BREthereumToken token) {
+    free (token->address);
+    free (token->symbol);
+    free (token->name);
+    free (token->description);
+    free (token);
+}
+
+extern void
+tokenUpdate (BREthereumToken token,
+             const char *symbol,
+             const char *name,
+             const char *description,
+             int decimals,
+             BREthereumGas defaultGasLimit,
+             BREthereumGasPrice defaultGasPrice) {
+
+    if (0 != strcasecmp (symbol     , token->symbol     )) { free (token->symbol     ); token->symbol      = strdup (symbol     ); }
```

Comment:
General question; can these (name, description, symbol) be unicode? The standard `strxyz` don't support multibyte encodings, IIRC.

Reply:
I've not thought about it.  Worthy of a JIRA to capture?

Reply:
Absolutely.